joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -32,6 +44,13 @@ trait AstCreatorHelper(implicit withSchemaValidation: ValidationMode) { this: As
       case _                             => nameFromNode(node)
   }
 
+  protected def getTypeFullNameFromAstNode(ast: Seq[Ast]): String = {
```

Comment:
I think this function is a bad idea, I would rather remove this and let the type propagation handle this kind of type validation. It may lead to unintended behaviour later on, and I think we should aim to use `AstCreatorHelper.nodeTypeFullName` as far as possible.

Reply:
Okay, this reference was from `gosrc2cpg`. The main reason is, the `AstCreatorHelper.nodeTypeFullName` takes in a `DotNetNodeInfo`, and I wanted to pass in an AST. What could be a workaround for that?